### PR TITLE
Include MailHog in Docker development environment

### PR DESCRIPTION
**Pull request by TomAbel**
I had some trouble getting MailHog running with Magento on the Docker environment. I think it's beneficial to add it to the repo so the next developer can quickly and easily test out sending mail.

What I did:
- MailHog container in docker-compose;
- Installing mhsendmail in Dockerfile;
- Use mhsendmail, MailHog container as SMTP server in PHP config;
- A section about configuring Magento to use MailHog in CONTRIBUTING.md.

I also specified the Docker image tag (vs latest) and fixed any missing periods and spelling mistakes.

### CODE REVIEW

**Review by kaittodesk — CHANGES_REQUESTED** [review]

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -67,9 +67,18 @@ The Smaily installation script uses [Allow Symlinks](https://github.com/sreichel
...
+```
+
+### Sending mail via MailHog
+
+The development environment has included MailHog as an SMTP server. This will enable the Magento store to send out emails. MailHog has a web interface, so you can view mail which has been sent. The web interface is available at <http://localhost:8025>.
+
+**Set SMTP server to MailHog**
+
+To make Magento use MailHog, you must go to the admin panel and change the mail sending host and port.
+Go to ```System > Configuration > System > Mail Sending Settings``` and change `host` to `mailhog`, and `port` to `1025`.
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply CONTRIBUTING.md]: Tested it out and it works without this step indeed. 5d1534d

File: .docker/sendmail-mailhog.ini [hunk 2]
```diff
@@ -0,0 +1 @@
+sendmail_path = '/usr/local/bin/mhsendmail --smtp-addr="mailhog:1025"'
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply .docker/sendmail-mailhog.ini]: 739a7d0

File: .docker/docker-compose.yml [hunk 3]
```diff
@@ -44,6 +45,12 @@ services:
     volumes:
       - magento1-db-data:/var/lib/mysql
 
+  mailhog:
+    image: mailhog/mailhog
+    ports:
+      - 1025:1025
+      - 8025:8025
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply .docker/docker-compose.yml]: 783d37c

File: .docker/Dockerfile [hunk 4]
```diff
@@ -17,5 +17,10 @@ RUN if grep -q ${group} /etc/group; then \
...
     && useradd -m -u ${user_id} -g ${group_id} ${username} -s /bin/bash
 
-# Add local www-data group to the user
+# Add local www-data group to the user.
 RUN usermod -a -G www-data ${username}
+
+# Install sendmail for MailHog.
+RUN wget https://github.com/mailhog/mhsendmail/releases/download/v0.2.0/mhsendmail_linux_amd64 \
+   && chmod +x mhsendmail_linux_amd64 \
+   && mv mhsendmail_linux_amd64 /usr/local/bin/mhsendmail
```
[the remark this answers is not in the document]
Comment (TomAbel) [reply .docker/Dockerfile]: Didn't notice to move it there. Added 9eae1dd

File: .docker/sendmail-mailhog.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (kaittodesk) [reply .docker/sendmail-mailhog.ini]: You changed the file name, but didn't update the `.docker/docker-compose.yml` file.

File: .docker/docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (kaittodesk) [reply .docker/docker-compose.yml]: Better, but change it to `magento_mailhog` for consistency with other containers.

File: .docker/sendmail-mailhog.ini (hunk 2 above)
[the remark this answers is not in the document]
Comment (TomAbel) [reply .docker/sendmail-mailhog.ini]: fixed 6d7a733

File: .docker/docker-compose.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (TomAbel) [reply .docker/docker-compose.yml]: set it to `magento1_mailhog` e8e5a42

**Review by kaittodesk — APPROVED** [review]

**Pull request merged**